app.py

```diff
@@ -0,0 +1,267 @@
+import os
+import datetime
+
+from cs50 import SQL
+from flask import Flask, flash, jsonify, redirect, render_template, request, session
+from flask_session import Session
+from tempfile import mkdtemp
+from werkzeug.exceptions import default_exceptions, HTTPException, InternalServerError
+from werkzeug.security import check_password_hash, generate_password_hash
+from datetime import date
+
+from helpers import apology, login_required, merc, response, apodlinks, getjoke
+
+# Configure application
+app = Flask(__name__)
+
+# Ensure templates are auto-reloaded
+app.config["TEMPLATES_AUTO_RELOAD"] = True
+
+# Ensure responses aren't cached
+@app.after_request
+def after_request(response):
+    response.headers["Cache-Control"] = "no-cache, no-store, must-revalidate"
+    response.headers["Expires"] = 0
+    response.headers["Pragma"] = "no-cache"
+    return response
+
+
+# Configure session to use filesystem (instead of signed cookies)
+app.config["SESSION_FILE_DIR"] = mkdtemp()
+app.config["SESSION_PERMANENT"] = False
+app.config["SESSION_TYPE"] = "filesystem"
+Session(app)
+
+# Configure CS50 Library to use SQLite database
+db = SQL("sqlite:///database.db")
```

Comment:
this is what i was talking about before:
everything that fetches or updates data from somewhere should be in its own module. or better yet - each data source its module.
for example - sql_store, movies_api_store, astrology_api_store etc...
and sweep all the nitty details of connection and error handling inside
so in this file you get a much cleaner code

but i think this would be out of scope now and should be a different branch and another PR